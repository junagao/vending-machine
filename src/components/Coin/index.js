export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
