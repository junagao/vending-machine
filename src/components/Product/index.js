export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
