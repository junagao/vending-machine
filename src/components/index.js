export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
