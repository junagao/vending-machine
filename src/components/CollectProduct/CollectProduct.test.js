import React from 'react';
import { shallow } from 'enzyme';
import { CollectProduct } from 'components';

describe('CollectProduct', () => {
  let wrapper;
  const selectedProductNameMock = 'coke';
  const selectedProductImgMock = 'images/coke.png';
  const isCollectedMock = true;

  beforeEach(() => {
    wrapper = shallow(
      <CollectProduct
        selectedProductName={selectedProductNameMock}
        selectedProductImg={selectedProductImgMock}
        onCollectProduct={jest.fn()}
        isCollected={isCollectedMock}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one div CollectProductContainer element', () => {
    expect(wrapper.find('CollectProductContainer').length).toEqual(1);
  });

  it('should not render ProductImg if isCollected is true', () => {
    expect(wrapper.find('ProductImg').length).toEqual(0);
  });

  it('should render one img ProductImg element with the correct `src, width, alt` props, if isCollected is false', () => {
    wrapper.setProps({ isCollected: false });
    expect(wrapper.find('ProductImg').length).toEqual(1);
    expect(wrapper.find('ProductImg').prop('src')).toEqual(`images/coke.png`);
    expect(wrapper.find('ProductImg').prop('width')).toEqual(`35`);
    expect(wrapper.find('ProductImg').prop('alt')).toEqual(`coke`);
  });
});
