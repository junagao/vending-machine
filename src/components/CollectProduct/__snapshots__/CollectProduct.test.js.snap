// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectProduct should render correctly given the required props 1`] = `<CollectProductContainer />`;
